qiskit_optimization/converters/linear_equality_to_penalty.py

```diff
@@ -36,11 +36,12 @@ def __init__(self, penalty: Optional[float] = None) -> None:
         """
         Args:
             penalty: Penalty factor to scale equality constraints that are added to objective.
-                     If None is passed, penalty factor will be automatically calculated.
+                     If None is passed, a penalty factor will be automatically calculated on
+                     every conversion.
         """
         self._src = None  # type: Optional[QuadraticProgram]
         self._dst = None  # type: Optional[QuadraticProgram]
-        self._penalty = penalty  # type: Optional[float]
+        self.penalty = penalty  # type: Optional[float]
```

Comment:
You don't need to change it to public.

Reply:
I am not changing it to public, I am using the setter when caling `self.penalty` (because the setter updates `_should_define_penalty`)

Reply:
I mean that shouldn't it be `self._penalty` instead of `self.penalty`.

Reply:
I think I get what you are saying, but removing the `_` is not making the variable public (or creating a new one) because assigning a value to `penalty` (like `self.penalty = something`) is actually calling the setter as defined with `@penalty.setter`.

Reply:
Ah, I got it. The line is actually calling `@penalty.setter`.